sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/generators.scala

```diff
@@ -94,6 +96,61 @@ case class UserDefinedGenerator(
 }
 
 /**
+ * Separate v1, ..., vk into n rows. Each row will have k/n columns. n must be constant.
+ * {{{
+ *   SELECT stack(2, 1, 2, 3)) ->
+ *   1      2
+ *   3      NULL
+ * }}}
+ */
+@ExpressionDescription(
+  usage = "_FUNC_(n, v1, ..., vk) - Separate v1, ..., vk into n rows.",
+  extended = "> SELECT _FUNC_(2, 1, 2, 3);\n  [1,2]\n  [3,null]")
+case class Stack(children: Seq[Expression])
+    extends Expression with Generator with ImplicitCastInputTypes with CodegenFallback {
+
+  override def inputTypes: Seq[DataType] =
+    Seq(IntegerType) ++ Seq.fill(children.length - 1)(children.tail.head.dataType)
+
+  override def checkInputDataTypes(): TypeCheckResult = {
```

Comment:
e.g. what if the first argument is not int type? and I'm also surprised that `stack(1, 1.0, 2)` works, we will cast `1.0` to int type, according to the definition of `inputTypes`